### PR TITLE
[TestCleanup] Clean up mock_engine and its test

### PR DESCRIPTION
**Pull request by wyzhang**
* Change its prefill result to mirror maxtext engine prefill result
    * Remove no longer needed code
    * Rename variables to be more descriptive
    * Clean up test to simplify the code

### CODE REVIEW

File: jetstream/engine/mock_engine.py [hunk 1]
```diff
@@ -31,19 +31,33 @@
...
+
+@struct.dataclass
+class Prefix:
+  """Result of the prefill step.
+
+  This structure currently matches the prefix structure returned
+  by the MaxText prefill operation, but it is flexible and can vary
+  depending on the engine implementation.
+  """
+  logits: Optional[jax.Array] = None
```
Comment (deqiangc) [inline jetstream/engine/mock_engine.py:56]: more of a question: is there a case that logits will be optional?

Comment (deqiangc) [reply jetstream/engine/mock_engine.py:56]: This is for mock_engine, so I guess this is possible for test only.

**Review by deqiangc — COMMENTED** [review]
LGTM

File: jetstream/tests/engine/test_mock_engine.py [hunk 2]
```diff
@@ -32,105 +32,107 @@
...
 
-  def _prefill_np(self):
+  def _prefill_nopadding(self):
     """Performs prefill and returns a kv cache."""
-    engine, params = self._setup()
     # A 2 will be pre-pended as 'bos' token from the vocab.
     text = "AB"
-    metadata = engine.get_tokenizer()
-    tokenizer = engine.build_tokenizer(metadata)
+    metadata = self._engine.get_tokenizer()
```
[the remark this answers is not in the document]
Comment (deqiangc) [reply jetstream/tests/engine/test_mock_engine.py:63]: also there is lint error. pls fix

[the remark this answers is not in the document]
Comment (wyzhang) [reply jetstream/tests/engine/test_mock_engine.py]: Done

**Review by deqiangc — APPROVED** [review]
